pkg/skaffold/deploy/labels.go

```diff
@@ -47,26 +48,27 @@ const (
 	sleeptime = 300 * time.Millisecond
 )
 
-func labelDeployResults(labels map[string]string, results []Artifact) error {
-	if len(labels) == 0 {
-		return nil
-	}
-
+func labelDeployResults(labels map[string]string, results []Artifact) (kubectl.Resources, error) {
 	// use the kubectl client to update all k8s objects with a skaffold watermark
 	dynClient, err := kubernetes.DynamicClient()
 	if err != nil {
-		return fmt.Errorf("error getting Kubernetes dynamic client: %w", err)
+		return nil, fmt.Errorf("error getting Kubernetes dynamic client: %w", err)
 	}
 
 	client, err := kubernetes.Client()
 	if err != nil {
-		return fmt.Errorf("error getting Kubernetes client: %w", err)
+		return nil, fmt.Errorf("error getting Kubernetes client: %w", err)
 	}
 
+	var resources kubectl.Resources
+
 	for _, res := range results {
```

Comment:
Since this loops `[]Artifacts` which only contains skaffold-built images, this still won't select containers for pre-built images which belong to a different Deployment altogether; it only improves upon init containers and multi-container pods, right?

Reply:
exactly. Not yet.